src/main/java/org/tugraz/sysds/api/DMLScript.java

```diff
@@ -157,15 +157,7 @@ public static void main(String[] args)
 	{
 		Configuration conf = new Configuration(ConfigurationManager.getCachedJobConf());
 		String[] otherArgs = new GenericOptionsParser(conf, args).getRemainingArgs();
-		
-		try {
-			DMLScript.executeScript(conf, otherArgs);
-		} catch (ParseException pe) {
-			System.err.println(pe.getMessage());
-		} catch (DMLScriptException e){
-			// In case of DMLScriptException, simply print the error message.
-			System.err.println(e.getMessage());
-		}
+		DMLScript.executeScript(conf, otherArgs);
```

Comment:
The Point here was that this Exception was already handled just in the next function, executeScript() and then thrown again just wrapped in another exception. That is why i moved the DMLScriptException. (look at line 239)
The other exception have exactly the same action, therefore it was moved as well. Since we are using Java Language level 8 we are able to catch multiple specific exceptions in a single catch block.

All that said, i can see i did not finish the work properly, since i am still throwing the exception, in the next method. (which is not needed anymore for same execution)

Reply:
Looking further into it, it seems like this part of the code could be cleaned up further in the exception handling, since there is many inconsistencies.